### PR TITLE
[SPARK-31495][SQL][FOLLOW-UP][3.0] Fix test failure of explain-aqe.sql

### PR DESCRIPTION
**Pull request by Ngone51**
### What changes were proposed in this pull request?

update the `explain-aqe.sql.out`

### Why are the changes needed?

Fix test failure.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

Updated test result file.

### CODE REVIEW

**Comment by Ngone51** [comment]
cc @dongjoon-hyun @cloud-fan @HyukjinKwon

**Review by HyukjinKwon — APPROVED** [review]
LGTM if tests pass

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thanks, @Ngone51 and @HyukjinKwon .
I verified the update test suite locally. Merged to branch-3.0 to recover Jenkins.

**Comment by Ngone51** [comment]
thanks!

File: sql/core/src/test/resources/sql-tests/results/explain-aqe.sql.out [hunk 1]
```diff
@@ -314,7 +314,7 @@ Arguments: HashedRelationBroadcastMode(List(cast(input[0, int, true] as bigint))
 Left keys [1]: [key#x]
 Right keys [1]: [key#x]
 Join condition: None
-
```
Comment (cloud-fan) [inline sql/core/src/test/resources/sql-tests/results/explain-aqe.sql.out:317]: does it mean the master branch outputs some extra spaces in EXPLAIN FORMATTED?

Comment (Ngone51) [reply sql/core/src/test/resources/sql-tests/results/explain-aqe.sql.out:317]: It turns out that we misused `stripMargin` for explained nodes. For example, in the case of `Join`,

https://github.com/apache/spark/blob/f543d6a1ee76de8cae417ff480fa9c0e0ce5343d/sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BaseJoinExec.scala#L50-L53

We actually should use `|""".stripMargin` to exclude the extra spaces.

I'll send a PR later to fix the issue.

**Pull request closed without merging**